src/engine/blocks.js

```diff
@@ -578,8 +584,10 @@ class Blocks {
 
             // Update block value
             if (!block.fields[args.name]) return;
-            if (args.name === 'VARIABLE' || args.name === 'LIST' ||
-                args.name === 'BROADCAST_OPTION') {
+            if ((args.name === 'VARIABLE' || args.name === 'LIST' ||
+                // Can get rid of the checks above and the && clause
+                // after extensionification of variables is complete
+                args.name === 'BROADCAST_OPTION') && block.fields[args.name].id) {
```

Comment:
The original condition here is for handling the old variable/list blocks. I added an extra check to make sure that the old blocks still work, and that this doesn't break the new extension-ified variable blocks. I expect this whole check to be removed when we are done extensionifying the Variables category.